libraries/AP_RangeFinder/AP_RangeFinder_Backend_CAN.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "AP_RangeFinder_Backend.h"
+
+#if HAL_MAX_CAN_PROTOCOL_DRIVERS
+
+#include <AP_CANManager/AP_CANSensor.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+class RangeFinder_MultiCAN;
+
+class AP_RangeFinder_Backend_CAN : public AP_RangeFinder_Backend
+{
+public:
+    // constructor
+    AP_RangeFinder_Backend_CAN(RangeFinder::RangeFinder_State &_state,
+                                  AP_RangeFinder_Params &_params);
+
+    friend class RangeFinder_MultiCAN;
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+protected:
+
+    // update state
+    virtual void update(void) override;
+
+    // get distance measurement
+    bool get_reading(float &reading_m);
+
+    // it is essential that anyone relying on the base-class update to implement this
+    virtual bool handle_frame(AP_HAL::CANFrame &frame) = 0;
+
+    // maximum time between readings before we change state to NoData:
+    virtual uint16_t read_timeout_ms() const { return 200; }
```

Comment:
We usually compare ms in uint32_t

Reply:
Would this make a difference? Timeout would usually be a few ms, maybe 50 - 500